### PR TITLE
Fix IncidentType Type

### PR DESCRIPTION
**Pull request by nVlast**
We recently encountered the following JSON unmarshal error while streaming events from the Event Streams API:
```
strconv.ParseUint: parsing "Unusual login to an endpoint": invalid syntax
```

I looked through the past issues in this repository and noticed that the `IncidentType` has changed a couple of times (from `*uint64` to `json.Number` to `*string` to `IntOrString`). Looking at the information provided by previous reporters and the error we're facing now, it looks like this field can be:
- an integer (e.g. `1`)
- an integer but in string form (e.g. `"1"`)
- an arbitrary string (e.g. `"Unusual login to an endpoint"`)

The integers seem to be used in "Incident Summary Event" events and the arbitrary strings seem to be used in "Identity Protection Incident" events.

### CODE REVIEW

File: falcon/models/streaming_models/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+type StringOrIntTest struct {
+	Value StringOrInt `json:"value"`
+}
+
+func TestStringOrInt(t *testing.T) {
+	testCases := []struct {
+		name           string
+		jsonData       string
+		expectedValue  string
```
Comment (kostaspap) [inline falcon/models/streaming_models/utils_test.go]: `expectedValue` seems redundant? `expectedOutput` is enough, right?

File: falcon/models/streaming_models/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		expectedOutput string
+	}{
+		{
+			name:           "Integer value",
+			jsonData:       `{"value": 42}`,
+			expectedValue:  "42",
```
Comment (kostaspap) [inline falcon/models/streaming_models/utils_test.go]: Can you add a test with `null` value?

File: falcon/models/streaming_models/models.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ type Event struct {
 	Tactic            *string           `json:"Tactic,omitempty"`
 	Technique         *string           `json:"Technique,omitempty"`
 	AuditKeyValues    *[]AuditKeyValues `json:"AuditKeyValues,omitempty"`
-	IncidentType      *IntOrString      `json:"IncidentType,omitempty"`
+	IncidentType      *StringOrInt      `json:"IncidentType,omitempty"`
```
Comment (kostaspap) [inline falcon/models/streaming_models/models.go]: Should we just call it `String`...? And readeers could just check the godoc to see the difference to `string`

**Review by kostaspap — CHANGES_REQUESTED** [review]

File: falcon/models/streaming_models/utils_test.go (hunk 1 above)
Comment (nVlast) [reply falcon/models/streaming_models/utils_test.go]: I wanted a way to check that unmarshaling leaves the correct values in the struct/memory, but you're probably right.

File: falcon/models/streaming_models/utils_test.go (hunk 2 above)
Comment (nVlast) [reply falcon/models/streaming_models/utils_test.go]: Good catch. Added.

File: falcon/models/streaming_models/models.go (hunk 3 above)
Comment (nVlast) [reply falcon/models/streaming_models/models.go]: Yeah, I'd go with anything, I couldn't find a good enough name anyway. Changed it to `String`.

**Comment by kostaspap** [comment]
I see that this repo does not run any unit tests by default. 
1. How easy/difficult is it to add a GHA action that runs the tests and blocks the merging if the tests fail? 
2. Have you run tests across the full repo?

**Comment by nVlast** [comment]
> I see that this repo does not run any unit tests by default.
> 
>     1. How easy/difficult is it to add a GHA action that runs the tests and blocks the merging if the tests fail?
> 
>     2. Have you run tests across the full repo?

1. I disabled manually (via GitHub's UI) all the inherited GitHub Actions workflows except the "build" one; I refined a bit and enabled it to run for us. I'll make it mandatory after we merge this PR (so that we have the refined version merged).
2. The only tests currently in the repo are the ones I'm adding in this PR, which I managed to run successfully:
```
gofalcon on  nvlast-fix-incidenttype-type via 🐹 took 6s 
❯ go test -v ./...
?   	github.com/crowdstrike/gofalcon/examples	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/container	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/customize_transport	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_cleanup_hosts	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_cleanup_pods	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_cspm_aws_accounts	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_cspm_ioms	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_custom_ioas	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_detection_details	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_discover_host_details	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_event_stream	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_get_cid	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_host_details	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_intel_indicators	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_intel_rules_download	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_iocs	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_registry_token	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_rtr_admin_create_and_run_script	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_rtr_batch_read_only_command	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_rtr_read_only_command	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_sensor_download	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_supported_kernels	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_vulnerabilities	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/falcon_zta	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/oauth_token	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/simple	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/simple_access_token	[no test files]
?   	github.com/crowdstrike/gofalcon/examples/stream_new_detections	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/a_s_p_m	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/alerts	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/api_integrations	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/certificate_based_exclusions	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cloud_aws_registration	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cloud_azure_registration	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cloud_connect_aws	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cloud_security_assets	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cloud_snapshots	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/compliance_assessments	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/configuration_assessment	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/configuration_assessment_evaluation_logic	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/container_alerts	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/container_detections	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/container_images	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/container_packages	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/container_vulnerabilities	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/content_update_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/correlation_rules	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cspg_iacapi	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/device_control_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/discover	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/discover_iot	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/cspm_registration	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/downloads_api	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/custom_ioa	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/event_streams	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/exposure_management	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/falcon_complete_dashboard	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/falcon_container	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/falcon_container_cli	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/device_content	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/drift_indicators	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/falconx_sandbox	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/delivery_settings	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/firewall_management	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/firewall_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/foundry_logscale	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/deployments	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/field_schema	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/filevantage	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/falcon_container_image	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/host_group	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/d4c_registration	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/handle	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/host_migration	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/hosts	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/detects	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/event_schema	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/datascanner	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/intelligence_feeds	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/intel	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/ioa_exclusions	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/custom_storage	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/image_assessment_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/iocs	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/kubernetes_protection	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/lookup_files	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/message_center	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/ml_exclusions	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/mobile_enrollment	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/humio_auth_proxy	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/oauth2	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/ods	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/overwatch_dashboard	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/prevention_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/quarantine	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/installation_tokens	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/quick_scan_pro	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/real_time_response	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/real_time_response_admin	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/real_time_response_audit	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/identity_protection	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/quick_scan	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/releases	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/report_executions	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/response_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/runtime_detections	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/installation_tokens_settings	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/sample_uploads	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/sensor_download	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/sensor_update_policies	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/sensor_usage_api	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/sensor_visibility_exclusions	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/incidents	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/spotlight_vulnerabilities	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/tailored_intelligence	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/threatgraph	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/unidentified_containers	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/ioc	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/vulnerabilities	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/workflows	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/zero_trust_assessment	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/models	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/identity_entities	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/spotlight_evaluation_logic	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/user_management	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/mssp	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/recon	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/scheduled_reports	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/release_notes	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/malquery	[no test files]
?   	github.com/crowdstrike/gofalcon/falcon/client/intelligence_indicator_graph	[no test files]
?   	github.com/crowdstrike/gofalcon/pkg/falcon_util	[no test files]
=== RUN   TestString
=== RUN   TestString/Integer_value
=== RUN   TestString/String_integer_value
=== RUN   TestString/Empty_string
=== RUN   TestString/Arbitrary_string
=== RUN   TestString/Float_value
=== RUN   TestString/Null_value
--- PASS: TestString (0.00s)
    --- PASS: TestString/Integer_value (0.00s)
    --- PASS: TestString/String_integer_value (0.00s)
    --- PASS: TestString/Empty_string (0.00s)
    --- PASS: TestString/Arbitrary_string (0.00s)
    --- PASS: TestString/Float_value (0.00s)
    --- PASS: TestString/Null_value (0.00s)
PASS
ok  	github.com/crowdstrike/gofalcon/falcon/models/streaming_models	0.476s
gofalcon on  nvlast-fix-incidenttype-type via 🐹 v1.23.8 took 9s 
❯ 
```

**Review by kostaspap — APPROVED** [review]
